sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -5243,11 +5243,11 @@ class AstBuilder extends DataTypeAstBuilder
       } else {
         None
       }
-      val setDefaultExpression: Option[String] =
+      val setDefaultExpression: Option[DefaultValueExpression] =
         if (action.defaultExpression != null) {
-          Option(action.defaultExpression()).map(visitDefaultExpression).map(_.originalSQL)
+          Option(action.defaultExpression()).map(visitDefaultExpression)
         } else if (action.dropDefault != null) {
-          Some("")
+          Some(DefaultValueExpression(Literal(""), ""))
```

Comment:
shall we use `DefaultValueExpression(null, "")` to indicate dropping default?

Reply:
sure, I was initially thinking Literal("") to match "" that was used before, but that makes sense too

Reply:
oh, how about `Literal(null)`? My point is that `Literal("")` means an empty string literal, which is `"''"`, not `""`

Reply:
ok, done